### PR TITLE
Add CFLAGS with -Werror for CI

### PR DESCRIPTION
**Pull request by bebehei**
This should fix dunst-project/docker-images#4

### CODE REVIEW

**Comment by bebehei** [comment]
As now seen in the CI: `-Werror` is used.

It's not possible to inject CFLAGS via environment in `make`, since this overwrites the default variable. :-/

I think this is more favorable over hardcoding `-Werror` in the docker images for testing.

File: config.mk [hunk 1]
```diff
@@ -6,6 +6,7 @@ DATADIR ?= ${PREFIX}/share
 # around for backwards compatibility
 MANPREFIX ?= ${DATADIR}/man
 MANDIR ?= ${MANPREFIX}
+CFLAGS_CI ?=
```
Comment (fwSmit) [inline config.mk:9]: I would call it `EXTRA_CFLAGS` and document it somewhere. It might be useful for other things as well.

**Comment by fwSmit** [comment]
I also thought about just adding -Werror to the DEFAULT_CFLAGS, but that might be a bit annoying in development.

File: config.mk (hunk 1 above)
Comment (bebehei) [reply config.mk]: I've renamed it to `EXTRA_CFLAGS`.

**Comment by bebehei** [comment]
Having `-Werror` hardcoded in the Makefile is very annoying, especially for users compiling the code on another distro, which potentially throws new errors.

**Comment by fwSmit** [comment]
> Having `-Werror` hardcoded in the Makefile is very annoying, especially for users compiling the code on another distro, which potentially throws new errors.

Agreed. 

By the way, I would expect at least the arch build to fail now, because on my system it gives a warning about a using deprecated function. This happens in a recent version of glib. Do the images have to be updated before the warning will show up?

**Comment by bebehei** [comment]
> Do the images have to be updated before the warning will show up?

Yes.

**Comment by fwSmit** [comment]
Am I able to update the images as well or do you need some sort of credentials?

**Review by fwSmit — APPROVED** [review]

**Comment by bebehei** [comment]
Afaik you have to have a docker hub account. I can add you to the orga then.

All you have to do, should be a `make build` locally and a `make push`.

It'll then push the freshly built images to docker hub.

**Comment by fwSmit** [comment]
Okay I made an account. My username is `fwsmit`. I don't think I'll need access immediately, but it'll be useful to be able to fix it when things go wrong. Also, I've made a fix for the warning from glib: 1d3822ee8b2263be472e13ea685e7beb69071fb1. Before that is merged the images shouldn't be updated.

**Comment by bebehei** [comment]
I've added you in the docker project. You're now an owner there. You can update it whenever you want. If you need help, I can give you suppoort.

**Pull request merged**